contractcourt/channel_arbitrator.go

```diff
@@ -1395,15 +1494,17 @@ func (c *ChannelArbitrator) channelAttendant(bestHeight int32) {
 				HtlcResolutions:  *closeInfo.HtlcResolutions,
 			}
 
-			// When processing a unilateral close event, we'll
-			// transition directly to the ContractClosed state.
-			// When the state machine reaches that state, we'll log
-			// out the set of resolutions.
 			stateCb := func(nextState ArbitratorState) error {
 				if nextState != StateContractClosed {
 					return nil
 				}
 
+				// When processing a unilateral close event,
+				// we'll transition to the ContractClosed
+				// state. When the state machine reaches that
+				// state, we'll log out the set of resolutions,
+				// such that they are available to fetch in
+				// that state.
```

Comment:
I don't think we need a callback anymore, we can just log them directly and close the channel.

Reply:
I agree! Since we know that we are eventually gonna end up in `StateContractClosed`, there's not really any reason to wait before logging the contract resolutions. Not sure why we were using state callbacks in the first place? 🤔 cc @roasbeef

That should simplify a lot of the logic surrounding the state machine a great deal! :tada: